lib/widgets/message_list.dart

```diff
@@ -366,14 +366,24 @@ class MarkAsReadWidget extends StatelessWidget {
       await showErrorDialog(context: context,
         title: zulipLocalizations.errorMarkAsReadFailedTitle,
         message: e.toString());
+      return;
+    }
+    if (!context.mounted) return;
+    if (narrow == const AllMessagesNarrow()) {
+      // We've learned that the mark-all-as-read work succeeded, so we know
+      // there can't be ancient unreads we don't know about. Update the
+      // Unreads model to reflect that.
+      //
+      // [Unreads] is supposed to do this naturally, in
+      // handleUpdateMessageFlagsEvent. That's based on the documented behavior
+      // that a completed mark-all-as-read task gets signaled by `all: true` on
+      // the [UpdateMessageFlagsAddEvent]. But when using this new batch-
+      // processing protocol, it seems clients have to track progress and
+      // success themselves; the event from the last batch is unmarked.
+      // Discussion:
+      //   <https://chat.zulip.org/#narrow/stream/243-mobile-team/topic/flutter.3A.20Mark-as-read/near/1680275>
+      PerAccountStoreWidget.of(context).unreads.oldUnreadsMissing = false;
```

Comment:
`notifyListeners` isn't being called after mutating the `unreads` model here. I think there's an argument for not doing that here (as we aren't using this flag in the UI) but leaving a note to that effect might be good?

Reply:
Oh right, thanks for catching that! I think it'd be good to call `notifyListeners`. We might in the future use this flag in the UI; here's from the doc on `old_unreads_missing` in the `/register` response:

> […] When true, we recommend that clients display a warning, as they are likely to produce erroneous results until reloaded with the user having fewer than MAX_UNREAD_MESSAGES unread messages.

I'll send a revision with one way of making that happen.